### PR TITLE
Testing TokamakSource generates points correctly

### PR DESCRIPTION
**Pull request by LiamPattinson**
I've added a test and refactored `test_tokamak_source.py` to ensure `TokamakSource` generates points within the expected shape. This change brings in `hypothesis` as an additional testing requirement, which is used to automatically generate a wide range of possible tokamaks and ensure it works for more than the single example given previously. It also includes a general refactor to avoid the repeated lines of code.

I believe this goes some way towards solving issue https://github.com/fusion-energy/openmc-plasma-source/issues/34, but I'd like to go a bit further. For example, the inputs to `TokamakSource` aren't checked for validity in its `__init__` function. It would be quite straightforward to implement these checks and associated unit tests, and I could raise this as separate issue/pull request.

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
Sorry somehow I wasn't notified of this PR thanks @shimwell for pointing it out. I shall have a look asap!

File: tests/test_tokamak_source.py [hunk 1]
```diff
@@ -1,74 +1,117 @@
...
-    assert pytest.approx(sum(my_source.strengths), 1)
-
-
-def test_angles():
+    assert pytest.approx(sum(tokamak_source.strengths), 1)
+
+
+@given(tokamak_source=tokamak_source_strategy())
+@settings(max_examples=100, deadline=None)
+def test_source_locations_are_within_correct_range(tokamak_source):
```
Comment (RemDelaporteMathurin) [inline tests/test_tokamak_source.py:69]: Nice one! 

I don't see the shafranov shift in this test, shouldn't it appear somewhere?

File: tests/test_tokamak_source.py [hunk 2]
```diff
@@ -1,74 +1,117 @@
...
+        "ion_density_centre" : 1.09e20,
+        "ion_density_peaking_factor" : 1,
+        "ion_density_pedestal" : 1.09e20,
+        "ion_density_separatrix" : 3e19,
+        "ion_temperature_centre" : 45.9,
+        "ion_temperature_peaking_factor" : 8.06,
+        "ion_temperature_pedestal" : 6.09,
+        "ion_temperature_separatrix" : 0.1,
+        "mode" : "H",
+        "ion_temperature_beta" : 6,
```
[the remark this answers is not in the document]
Comment (shimwell) [reply tests/test_tokamak_source.py:46]: I thought we had a github action for pep8 on this repo

**Comment by shimwell** [comment]
I am not sure why the pep workflow called black didn't run on this

File: tests/test_tokamak_source.py (hunk 1 above)
Comment (LiamPattinson) [reply tests/test_tokamak_source.py:69]: I didn't think the Shafranov shift affected the shape of the last closed magnetic surface, as in the equations for RZ that term equals 0 when a=A, and this test only checks that R is within that outer bound.

**Comment by LiamPattinson** [comment]
Apologies for missing the PEP8 formatting, still trying to get in the habit of calling `black` regularly.

File: tests/test_tokamak_source.py (hunk 1 above)
Comment (RemDelaporteMathurin) [reply tests/test_tokamak_source.py:70]: You are right!

**Pull request merged**